keps/sig-node/2400-node-swap/README.md

```diff
@@ -183,14 +194,54 @@ Allocate the swap limit equal to the requested memory for each container and adj
+
+System critical daemons (such as Kubelet) are essential for node health. Usually, an appropriate portion of system resources (e.g., memory, CPU) is reserved as system reserved. However, swap doesn't inherently support reserving a portion out of the total available. For instance, in the case of memory, we set `memory.min` on the node-level cgroup to ensure an adequate amount of memory is set aside, away from the pods, and for system critical daemons. But there is no equivalent for swap; i.e., no `memory.swap.min` is supported in the kernel.
 
 Since this proposal advocates enabling swap only for the Burstable QoS pods, this can be done by setting `memory.swap.max` on the cgroups used by the Burstable QoS pods. The value of this `memory.swap.max` can be calculated by:
 
 memory.swap.max = total swap memory available on the system - system reserve (memory)
 
 This is the total amount of swap available for all the Burstable QoS pods; let's call it `TotalPodsSwapAvailable`. This will ensure that the system critical daemons will have access to the swap at least equal to the system reserved memory. This will indirectly act as having support for swap in system reserved.
 
+#### Best Practices
+
+This section is a recommendation for how to set up your nodes with swap if using this feature.
+
+##### Disable swap for system critical daemon
+
+As we were testing this feature, we found degration of services if you allow system critical daemons to swap.
+The recommendation is to set the cgroup for the system slice to avoid swap (ie `memory.swap.max 0`).
+While doing this, we found that it is still possible for workloads to impact critical daemons.
+
+##### Protect system critical daemons for iolatency
+
+As we disabled swap for system slice, we saw cases where the system.slice would still be impacted by workloads swapping.
+The workloads need to have less priority for IO than the system slice. We found that setting `io.latency` for system.slice fixes these issues.
+
+See [io-control](https://facebookmicrosites.github.io/cgroup2/docs/io-controller.html#protecting-workloads-with-iolatency) for more details.
+
+##### Control Plane Swap
+
+We only recommend enabling swap for the worker nodes. The control plane contains mostly Guaranteed QoS Pods, so swap may be disabled mostly.
+The main concern would be swapping in the critical services on the control plane which can cause a performance impact.
+
+##### Use of a dedicated disk for swap
+
+We recommend using a separate disk for your swap partition. With [encryption](#security-risk), an encrypted disk is recommended for swap.
+If swap is on a partition or the root filesystem, workloads can interfere with system processes needing to write to disk.
+If they occupy the same disk, it's possible processes can overwhelm swap and throw off the I/O of kubelet/cri-o/systemd, which would affect other workloads
```

Comment:
maybe sub CRI-O for `CRI implementation` or `container runtime`?

Reply:
updated.